apps/pkt-gen/pkt-gen.c

```diff
@@ -983,8 +1044,8 @@ initialize_packet(struct targ *targ)
 	}
 #endif
 
-	paylen = targ->g->pkt_size - sizeof(*eh) -
-	    (targ->g->af == AF_INET ? sizeof(ip): sizeof(ip6));
+	paylen = targ->g->pkt_size -
+	    PKT_OVERHD(targ->g->af, targ->g->virt_header);
```

Comment:
`pkt_size` does not include `virt_header`... why are you including that in `PKT_OVERHD`?